### ISSUE TITLE
Simplify and improve the s-expression parser

### ISSUE THREAD

**Issue by kasbah** [monostable/haskell-kicad-data#6]
S-expressions should be super simple to parse, they are a representation of the parse tree! 

1. Do we really need to be aware of keywords in the s-expression parser? 
2. It should output line number and character number along with the parse output so that we can report these if they can not be interpreted later

**Comment by mignon-p** [comment]
Yeah, I think the s-expression parser would be more general without keywords.  Right now, I can't use it to parse a `.kicad_pcb` file because I get this error:

```
Parse Error: "SExpr" (line 1, column 2):
unexpected "k"
expecting keyword
```

when I parse a file that starts like this:

```
(kicad_pcb (version 4) (host pcbnew 4.0.6)
```

**Comment by kasbah** [comment]
I think we should be able to define it as `AtomKey String` [here](https://github.com/monostable/haskell-kicad-data/blob/ed200c3e4f1f75798dad81eda78b960a0b8d4695/Data/Kicad/SExpr/SExpr.hs#L8) and then go from `String -> PcbnewExpr` [here](https://github.com/monostable/haskell-kicad-data/blob/ed200c3e4f1f75798dad81eda78b960a0b8d4695/Data/Kicad/PcbnewExpr/Parse.hs#L30). Might have to ditch the AtomDbl as well because the KiCad format is super lax about putting in quotes for strings (I think that's why I put the keywords in, to be able to tell if it's a number or a string from context).

**Comment by kasbah** [comment]
Fixed with #13
